tests fail on clean system

### What happened?

I created a clean Fedora 41 virtual machine, installed Go, checked out goreleaser, and ran Goreleaser's tests. The tests failed in the `internal/pipe/gomod` package:

```console
$ go test ./internal/pipe/gomod
--- FAIL: TestGoModProxy (4.32s)
    --- FAIL: TestGoModProxy/goreleaser (0.71s)
        gomod_proxy_test.go:119: 
                Error Trace:    /home/twp/src/github.com/goreleaser/goreleaser/internal/pipe/gomod/gomod_proxy_test.go:119
                Error:          Received unexpected error:
                                failed to proxy module: exit status 1: go: github.com/goreleaser/goreleaser@v0.161.1 indirectly requires nhooyr.io/websocket@v1.8.6: reading nhooyr.io/websocket/go.mod at revision v1.8.6: unknown revision v1.8.6
                Test:           TestGoModProxy/goreleaser
    --- FAIL: TestGoModProxy/no_perms (1.24s)
        --- FAIL: TestGoModProxy/no_perms/go.mod (0.41s)
            gomod_proxy_test.go:203: 
                        Error Trace:    /home/twp/src/github.com/goreleaser/goreleaser/internal/pipe/gomod/gomod_proxy_test.go:203
                        Error:          Received unexpected error:
                                        failed to proxy module: exit status 1: go: github.com/goreleaser/goreleaser@v0.161.1 indirectly requires nhooyr.io/websocket@v1.8.6: reading nhooyr.io/websocket/go.mod at revision v1.8.6: unknown revision v1.8.6
                        Test:           TestGoModProxy/no_perms/go.mod
        --- FAIL: TestGoModProxy/no_perms/go.sum (0.40s)
            gomod_proxy_test.go:203: 
                        Error Trace:    /home/twp/src/github.com/goreleaser/goreleaser/internal/pipe/gomod/gomod_proxy_test.go:203
                        Error:          Received unexpected error:
                                        failed to proxy module: exit status 1: go: github.com/goreleaser/goreleaser@v0.161.1 indirectly requires nhooyr.io/websocket@v1.8.6: reading nhooyr.io/websocket/go.mod at revision v1.8.6: unknown revision v1.8.6
                        Test:           TestGoModProxy/no_perms/go.sum
        --- FAIL: TestGoModProxy/no_perms/../../../go.sum (0.44s)
            gomod_proxy_test.go:203: 
                        Error Trace:    /home/twp/src/github.com/goreleaser/goreleaser/internal/pipe/gomod/gomod_proxy_test.go:203
                        Error:          Received unexpected error:
                                        failed to proxy module: exit status 1: go: github.com/goreleaser/goreleaser@v0.161.1 indirectly requires nhooyr.io/websocket@v1.8.6: reading nhooyr.io/websocket/go.mod at revision v1.8.6: unknown revision v1.8.6
                        Test:           TestGoModProxy/no_perms/../../../go.sum
    --- FAIL: TestGoModProxy/goreleaser_with_main.go (0.43s)
        gomod_proxy_test.go:237: 
                Error Trace:    /home/twp/src/github.com/goreleaser/goreleaser/internal/pipe/gomod/gomod_proxy_test.go:237
                Error:          Received unexpected error:
                                failed to proxy module: exit status 1: go: github.com/goreleaser/goreleaser@v0.161.1 indirectly requires nhooyr.io/websocket@v1.8.6: reading nhooyr.io/websocket/go.mod at revision v1.8.6: unknown revision v1.8.6
                Test:           TestGoModProxy/goreleaser_with_main.go
FAIL
FAIL    github.com/goreleaser/goreleaser/v2/internal/pipe/gomod 4.343s
FAIL
```

I expected the tests to pass.

### How can we reproduce this?

```console
$ docker run -it fedora:latest /bin/bash
[root@713dff0e0355 /]# dnf install git golang
...
[root@713dff0e0355 /]# git clone https://github.com/goreleaser/goreleaser.git
...
[root@713dff0e0355 /]# cd goreleaser
[root@713dff0e0355 goreleaser]# go test ./internal/pipe/gomod
...
--- FAIL: TestGoModProxy (114.17s)
    --- FAIL: TestGoModProxy/goreleaser (106.01s)
        gomod_proxy_test.go:119:
                Error Trace:    /goreleaser/internal/pipe/gomod/gomod_proxy_test.go:119
                Error:          Received unexpected error:
                                failed to proxy module: exit status 1: go: downloading github.com/goreleaser/goreleaser v0.161.1
                                go: github.com/goreleaser/goreleaser@v0.161.1 indirectly requires nhooyr.io/websocket@v1.8.6: reading nhooyr.io/websocket/go.mod at revision v1.8.6: unknown revision v1.8.6
                Test:           TestGoModProxy/goreleaser
...
    --- FAIL: TestGoModProxy/goreleaser_with_main.go (0.41s)
        gomod_proxy_test.go:237:
                Error Trace:    /goreleaser/internal/pipe/gomod/gomod_proxy_test.go:237
                Error:          Received unexpected error:
                                failed to proxy module: exit status 1: go: github.com/goreleaser/goreleaser@v0.161.1 indirectly requires nhooyr.io/websocket@v1.8.6: reading nhooyr.io/websocket/go.mod at revision v1.8.6: unknown revision v1.8.6
                Test:           TestGoModProxy/goreleaser_with_main.go
FAIL
FAIL    github.com/goreleaser/goreleaser/v2/internal/pipe/gomod 114.202s
FAIL
```

### goreleaser version

```bash
Latest `master` (f85e50fa218756e20a278a22629b9b9c6d268c25).
```


### GoReleaser Check

- [X] goreleaser check shows no errors

### Search

- [X] I did search for other open and closed issues before opening this

### Supporter

- [ ] I am a [sponsor](https://github.com/sponsors/caarlos0/) or a [Pro](https://goreleaser.com/pro) customer

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Additional context

I suspect, but have not verified, that these tests use a `go.mod` from a very old version of goreleaser ([`v0.161.1` to be precise](https://github.com/goreleaser/goreleaser/blob/main/internal/pipe/gomod/gomod_proxy_test.go#L116)). I suspect that this `go.mod` contains an indirect reference to the `nhooyr.io/websocket` module, which, as far as I can tell, no longer exists. I suspect that the tests pass on your machine and in CI because `nhooyr.io/websocket` is present in your Go module cache.

I tried changing the tag at https://github.com/goreleaser/goreleaser/blob/main/internal/pipe/gomod/gomod_proxy_test.go#L116 but encountered more errors:

* Using the `v1.0.0` tag produced the same errors.
* Using the last release of v1, i.e. `v1.9.2`, produced the same errors albeit referring to a slightly later version of `nhooyr.io/websocket` (1.8.7 instead of 1.8.6).
* Using any `v2` tag, e.g. the latest `v2.5.1`, produced Go module errors about Go module major versions.

--- a/internal/pipe/gomod/gomod_proxy_test.go
+++ b/internal/pipe/gomod/gomod_proxy_test.go
@@ -113,7 +113,7 @@ func TestGoModProxy(t *testing.T) {
 					Dir:    ".",
 				},
 			},
-		}, testctx.WithCurrentTag("v0.161.1"), withGoReleaserModulePath)
+		}, testctx.WithCurrentTag("v2.5.1"), withGoReleaserModulePath)
 
 		fakeGoModAndSum(t, ctx.ModulePath)
 		require.NoError(t, ProxyPipe{}.Run(ctx))
@@ -197,7 +197,7 @@ func TestGoModProxy(t *testing.T) {
 							Goarch: []string{runtime.GOARCH},
 						},
 					},
-				}, withGoReleaserModulePath, testctx.WithCurrentTag("v0.161.1"))
+				}, withGoReleaserModulePath, testctx.WithCurrentTag("v2.5.1"))
 
 				fakeGoModAndSum(t, ctx.ModulePath)
 				require.NoError(t, ProxyPipe{}.Run(ctx)) // should succeed at first
@@ -231,7 +231,7 @@ func TestGoModProxy(t *testing.T) {
 					Main:   "main.go",
 				},
 			},
-		}, withGoReleaserModulePath, testctx.WithCurrentTag("v0.161.1"))
+		}, withGoReleaserModulePath, testctx.WithCurrentTag("v2.5.1"))
 
 		fakeGoModAndSum(t, ctx.ModulePath)
 		require.NoError(t, ProxyPipe{}.Run(ctx))
@@ -306,7 +306,7 @@ func fakeGoMod(tb testing.TB, module string) {
 }
 
 func withGoReleaserModulePath(ctx *context.Context) {
-	ctx.ModulePath = "github.com/goreleaser/goreleaser"
+	ctx.ModulePath = "github.com/goreleaser/goreleaser/v2"
 }
 
 func withNfpmModulePath(ctx *context.Context) {

